provider/pkg/resources/suppress_diff.go

```diff
@@ -154,36 +163,449 @@ func getTagKey(tag resource.PropertyValue) string {
+// respecting CamelCase word boundaries. For example:
+// - "DbClusterIdentifier" -> "DBClusterIdentifier" (Db->DB, but Id in Identifier is not replaced)
+// - "IpProtocol" -> "IPProtocol"
+// - "ResourceId" -> "ResourceID"
+func replaceCamelCaseAcronyms(s string) string {
+	// Common acronyms that CloudFormation uses uppercase
+	acronyms := []struct {
+		lower string
+		upper string
+	}{
+		// Order matters: longer acronyms first to avoid partial matches
+		{"Https", "HTTPS"},
+		{"Http", "HTTP"},
+		{"Arn", "ARN"},
+		{"Kms", "KMS"},
+		{"Vpc", "VPC"},
+		{"Ec2", "EC2"},
+		{"Ssl", "SSL"},
+		{"Tls", "TLS"},
+		{"Cpu", "CPU"},
+		{"Ram", "RAM"},
+		{"Api", "API"},
+		{"Uri", "URI"},
+		{"Url", "URL"},
+		{"Dns", "DNS"},
+		{"Ssh", "SSH"},
+		{"Efs", "EFS"},
+		{"Rds", "RDS"},
+		{"Sns", "SNS"},
+		{"Sqs", "SQS"},
+		{"Iam", "IAM"},
+		{"Db", "DB"},
+		{"Ip", "IP"},
+		{"Id", "ID"},
+	}
+
+	result := s
+	for _, acr := range acronyms {
+		result = replaceAcronymAtWordBoundary(result, acr.lower, acr.upper)
+	}
+	return result
+}
+
+// replaceAcronymAtWordBoundary replaces an acronym only at CamelCase word boundaries.
+// A word boundary in CamelCase is:
+// - Start of string
+// - After a lowercase letter (e.g., "resourceId" - the 'I' starts a new word)
+// The acronym must be followed by:
+// - End of string
+// - An uppercase letter (next word)
+// - A non-letter (number, etc.)
+func replaceAcronymAtWordBoundary(s, lower, upper string) string {
+	if len(lower) == 0 {
+		return s
+	}
+
+	var result strings.Builder
+	result.Grow(len(s))
+
+	i := 0
+	for i < len(s) {
+		// Check if we can match the acronym at this position
+		if i+len(lower) <= len(s) && s[i:i+len(lower)] == lower {
+			// Check if this is a valid word boundary position
+			// The acronym should start at a word boundary (start or after lowercase)
+			atWordStart := i == 0 || isLower(s[i-1])
+
+			// The acronym should end at a word boundary
+			afterIdx := i + len(lower)
+			atWordEnd := afterIdx >= len(s) || isUpper(s[afterIdx]) || !isLetter(s[afterIdx])
+
+			if atWordStart && atWordEnd {
+				result.WriteString(upper)
+				i += len(lower)
+				continue
+			}
+		}
+		result.WriteByte(s[i])
+		i++
+	}
+
+	return result.String()
+}
+
+func isLower(b byte) bool {
+	return b >= 'a' && b <= 'z'
+}
+
+func isUpper(b byte) bool {
+	return b >= 'A' && b <= 'Z'
+}
+
+func isLetter(b byte) bool {
+	return isLower(b) || isUpper(b)
+}
+
+// propertyMapToMap converts a Pulumi PropertyMap to a standard Go map.
+func propertyMapToMap(pm resource.PropertyMap) map[string]interface{} {
+	if pm == nil {
+		return nil
+	}
+	result := make(map[string]interface{}, len(pm))
+	for k, v := range pm {
+		result[string(k)] = propertyValueToInterface(v)
+	}
+	return result
+}
+
+// propertyValueToInterface converts a Pulumi PropertyValue to a standard Go interface.
+func propertyValueToInterface(pv resource.PropertyValue) interface{} {
```

Comment:
I'm not sure these two functions are needed. It looks like when we use `propertyValueToInterface` by itself it is eventually converted back to a `map[string]interface{}`. I think we might just want to replace `propertyMapToMap` with `pm.Mappable` which produces a `map[string]any`.